### PR TITLE
Introduce maven:dependency:add

### PR DESCRIPTION
**Pull request by iocanel**
The pull request addresses some issues with the `quarkus:app:create` action and introduces `maven:dependency:add`.

### CODE REVIEW

File: packages/backend/package.json [hunk 1]
```diff
@@ -39,9 +39,10 @@
     "@backstage/plugin-search-backend-module-techdocs": "^0.1.23",
     "@backstage/plugin-search-backend-node": "^1.2.23",
     "@backstage/plugin-techdocs-backend": "^1.10.5",
+    "@qshift/plugin-maven-backend": "^0.1.28",
```
Comment (cmoulliard) [inline packages/backend/package.json:42]: Should we continue to use the suffix `*-backend` for code handling actions ? This is something that we should fix in a separate PR as a ticket exists

File: packages/backend/src/plugins/scaffolder.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ import type { PluginEnvironment } from '../types';
 import { ScmIntegrations } from '@backstage/integration';
 import { createBuiltinActions, createRouter } from '@backstage/plugin-scaffolder-backend';
 import { createQuarkusApp, cloneQuarkusQuickstart } from '@qshift/plugin-quarkus-backend';
+import { mavenDependencyAdd } from '@qshift/plugin-maven-backend';
 import { createArgoCdResources } from '@roadiehq/scaffolder-backend-argocd'
 
 export default async function createPlugin(
```
Comment (cmoulliard) [inline packages/backend/src/plugins/scaffolder.ts:10]: This code will be refactored to be aligned with the new backend. I already did it for ArgoCD (see PR - https://github.com/q-shift/backstage-playground/pull/89/files#diff-81c32eb256d5cca74cda7f7a2b4e9ed7dfe73061f61a29f5a49c031517e27d96).

File: plugins/maven-backend/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# maven
```
Comment (cmoulliard) [inline plugins/maven-backend/README.md]: The README should be reviewed to document how to use it. Example: https://github.com/q-shift/backstage-playground/blob/main/plugins/quarkus-backend/README.md

File: plugins/maven-backend/src/scaffolder/actions/maven/dependencyAdd.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    async handler(ctx) {
+        const pomPath = resolveSafeChildPath(ctx.workspacePath, ctx.input.values.pomPath ?? 'pom.xml');
+        if (!fs.existsSync(pomPath)) {
+          throw new Error(`File ${pomPath} not found`);
+        }
+        
+        const pomContent = fs.readFileSync(pomPath, 'utf8');
+        const pomXml = new DOMParser().parseFromString(pomContent, 'text/xml');
+        const project = pomXml.documentElement;
+        const dependencies = Array.from(project.childNodes).find((node: any) => node.nodeName === 'dependencies')
```
Comment (cmoulliard) [inline plugins/maven-backend/src/scaffolder/actions/maven/dependencyAdd.ts]: This code checks if `<dependencies/>` exists and if empty then `<dependencies> tag will be created BUT what about if the dependency that you would like to add already exists under `<dependencies/>` ?

File: packages/backend/package.json (hunk 1 above)
Comment (iocanel) [reply packages/backend/package.json:42]: The suffix is generated by the `yarn new --select backend-plugin` so I would keep it.

File: packages/backend/src/plugins/scaffolder.ts (hunk 2 above)
Comment (iocanel) [reply packages/backend/src/plugins/scaffolder.ts:10]: This is not something that *needs* to be done as part of this pull request.

Comment (cmoulliard) [reply packages/backend/src/plugins/scaffolder.ts:10]: Yep but the folder packaging actions could be named `plugin-maven-actions`, `plugin-quarkus-actions`, etc

File: plugins/maven-backend/README.md (hunk 3 above)
Comment (iocanel) [reply plugins/maven-backend/README.md]: Added doc

File: plugins/maven-backend/src/scaffolder/actions/maven/dependencyAdd.ts (hunk 4 above)
Comment (iocanel) [reply plugins/maven-backend/src/scaffolder/actions/maven/dependencyAdd.ts]: The plugin does not check if dependency already exists. This is something we can add later on.

Comment (cmoulliard) [reply plugins/maven-backend/src/scaffolder/actions/maven/dependencyAdd.ts]: "The plugin does not check if dependency already exists. This is something we can add later on." Can you then create a ticket ?

File: packages/backend/src/plugins/scaffolder.ts (hunk 2 above)
Comment (iocanel) [reply packages/backend/src/plugins/scaffolder.ts:10]: I don't understand what you are trying to say. The code is aligned with what we have in main.

Comment (cmoulliard) [reply packages/backend/src/plugins/scaffolder.ts:10]: My point is related to this ticket: https://github.com/q-shift/backstage-plugins/issues/9

File: plugins/maven-backend/src/scaffolder/actions/maven/dependencyAdd.ts (hunk 4 above)
Comment (iocanel) [reply plugins/maven-backend/src/scaffolder/actions/maven/dependencyAdd.ts]: I'll create a ticket once we get this merged.

**Review by cmoulliard — APPROVED** [review]

**Pull request merged**